include/jemalloc/internal/arena_inlines_a.h

```diff
@@ -34,44 +32,24 @@ arena_internal_get(arena_t *arena) {
 }
 
 JEMALLOC_INLINE bool
-arena_prof_accum_impl(arena_t *arena, uint64_t accumbytes) {
-	cassert(config_prof);
-	assert(prof_interval != 0);
-
-	arena->prof_accumbytes += accumbytes;
-	if (arena->prof_accumbytes >= prof_interval) {
-		arena->prof_accumbytes %= prof_interval;
-		return true;
-	}
-	return false;
-}
-
-JEMALLOC_INLINE bool
-arena_prof_accum_locked(arena_t *arena, uint64_t accumbytes) {
-	cassert(config_prof);
-
-	if (likely(prof_interval == 0)) {
-		return false;
-	}
-	return arena_prof_accum_impl(arena, accumbytes);
-}
-
-JEMALLOC_INLINE bool
-arena_prof_accum(tsdn_t *tsdn, arena_t *arena, uint64_t accumbytes) {
+arena_prof_accum(arena_t *arena, uint64_t accumbytes) {
 	cassert(config_prof);
 
 	if (likely(prof_interval == 0)) {
 		return false;
 	}
 
-	{
-		bool ret;
-
-		malloc_mutex_lock(tsdn, &arena->lock);
-		ret = arena_prof_accum_impl(arena, accumbytes);
-		malloc_mutex_unlock(tsdn, &arena->lock);
-		return ret;
-	}
+	uint64_t a0, a1;
+	bool overflow;
+	do {
+		a0 = atomic_read_u64(&arena->prof_accumbytes);
+		a1 = a0 + accumbytes;
+		overflow = (a1 >= prof_interval);
```

Comment:
Is it possible for accumbytes to be huge? If so this may overflow uint64, in which case the overflow calculation needs to be adjusted.